include/revng/BasicAnalyses/MaterializedValue.h

```diff
@@ -31,13 +31,17 @@ class MaterializedValue {
 
 public:
   bool operator==(const MaterializedValue &Other) const {
-    auto This = std::tie(IsValid, SymbolName, Value);
-    auto That = std::tie(Other.IsValid, Other.SymbolName, Other.Value);
+    auto MaxBitWidth = std::max(Value.getBitWidth(), Other.Value.getBitWidth());
+    auto ZExtValue = Value.zextOrSelf(MaxBitWidth);
+    auto ZExtOther = Other.Value.zextOrSelf(MaxBitWidth);
+    auto This = std::tie(IsValid, SymbolName, ZExtValue);
+    auto That = std::tie(Other.IsValid, Other.SymbolName, ZExtOther);
     return This == That;
   }
```

Comment:
Maybe we should `assert` here and let the caller take care of this.
I can look into this.

Reply:
Ok, just be aware that llvm does the assertion on its own already. I decided to fix it because it triggered failures at runtime in `revng` tests.

IMHO `operator==` is tricky, because it might be used under the hood in some algorithm that uses `MaterializedValue`s. My feeling was that in this case it was easier to fix `operator==` than force all users to be aware of this. For instance, what if someone puts a bunch of `MaterializedValue`s into a `std::vector` and calls `std::find` on it? It looks an innocuous operation, but if `operator==` calls `assert` under the hood it might crash.

Just my 2 cents. If you come up with a different solution let me know.